### ISSUE TITLE
Embed code

### ISSUE THREAD

**Issue by abhijeet9920** [geongeorge/i-hate-regex#55]
Hello,
I'm working on this repository for adding some regex which I thought may be useful. I followed all the mentioned steps. However I'm not getting proper embed turing machine flow.
For my regex it is showing invalid embed image. You can see it [here](https://embed.ihateregex.io/make/JTVFJTVCQS1aJTVEJTdCNSU3RCU1QjAtOSU1RCU3QjQlN0QlNUJBLVolNUQlN0IxJTdEJTI0).
Is there anything I'm doing wrong?
Thank you

**Comment by geongeorge** [comment]
Hey @abhijeet9920 ,
Thanks for your willingness in helping with adding expressions.

As far as I can see the link you provided shows a proper regular expression visualization.
I was also able to embed the url as you see [here](https://jsfiddle.net/zkh2r1uw/)

I assume that you are trying to embed the expression as an image which is not the way to go.
You have to use an iframe to embed the page onto your website. 

Just click the **</> embed code**  next to the visualization to copy the code.
And paste it as html as it contains the required <iframe> tags

**Example:**
``` htm
<iframe
src="https://embed.ihateregex.io/make/JTVFJTVCJTVDJTVDJTJCJTVEJTNGJTVCKCU1RCUzRiU1QjAtOSU1RCU3QjMlN0QlNUIpJTVEJTNGJTVCLSU1QyU1Q3MlNUMlNUMuJTVEJTNGJTVCMC05JTVEJTdCMyU3RCU1Qi0lNUMlNUNzJTVDJTVDLiU1RCUzRiU1QjAtOSU1RCU3QjQlMkM2JTdEJTI0"
height="300" style="width:100%"  frameborder="0"  />      
```

**Comment by abhijeet9920** [comment]
Hey @geongeorge thank you for your reply. 

**Example:**
```php
/*
Below is my used regex 
*/
$regex = '^([A-Z]{5})([0-9]{4})([A-Z]{1})$';
```
 While Embedded image is referring
```php
$regex = '^[A-Z]{4}[0-9]{3}[A-Z]$';
```
This is what I understood.
Do we need to add our regex to any other json apart from static.json file? From where this **embed code** is generating?

**Comment by geongeorge** [comment]
Hey mate,
I understand your problem. 
This is something that people often mistake.

(I'm assuming that you're getting [this embed](https://jsfiddle.net/nbfsy16t/) )

I hope this will help:

### How to read regex diagrams

These diagrams are similar(but not the same) to [state diagrams](https://en.wikipedia.org/wiki/State_diagram).

Imagine a simple diagram for the expression
```
^[a-z]{0,2}$
```
![image](https://user-images.githubusercontent.com/29296982/76619526-1e1a0700-6551-11ea-8af0-1c5760fe2814.png)


To read this:
1. Imagine you're starting from the **leftmost dot**
2. You pass through the **Start of line** character **^** (once)
3. You reach a junction where you can take 2 routes
    1. You Pass through the **[a-z]** (once)
        1. You can also repeat and go through **[a-z]** again. (at most: once)
    2. You're allowed to skip the **[a-z]** (0 times)
4. You pass through the **End of line** character **$** (once)
5. You reach the end

Now although the diagram shows **at most once**  next for the pattern **[a-z]** it means you pass through it once and you are allowed to repeat it once more adding up to 2 just like in the expression.


### TL;DR

For the expression:  `^[a-z]{0,2}$`

![image](https://user-images.githubusercontent.com/29296982/76619526-1e1a0700-6551-11ea-8af0-1c5760fe2814.png)

Although the diagram shows **at most once**  next for the pattern **[a-z]** it means you pass through it once and you are allowed to repeat it once more adding up to 2 just like in the expression.




>  Sorry for the long comment. I thought I'll mention this somewhere so I can link to this in the future

**Comment by geongeorge** [comment]
I'm closing this now. Feel free to open a new issue if you still have problemss
